### PR TITLE
Vue

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -106,28 +110,37 @@ api.optimize = function(options, callback) {
...
-      // FIXME: currently just supporting one static output name
-      path: path.dirname(options.output),
+      path: paths.local,
+      publicPath: paths.public,
       filename: path.basename(options.output)
     },
     resolve: {
       modules: [
         // top level node_modules dir
         // FIXME: get this via location independent method
```
Comment (mattcollier) [inline lib/index.js]: Is this still applicable?

**Comment by mattcollier** [comment]
Looks like this needs to be rebased, shows conflics in changelog.

**Comment by davidlehn** [comment]
If we start using this is many non-vue projects, then might want to pull out vue support into it's own package.  I think it doesn't have any overhead if you don't use .vue files?

File: package.json [hunk 2]
```diff
@@ -25,13 +25,19 @@
...
     "babel-runtime": "^6.23.0",
+    "css-loader": "^0.28.11",
     "file-size": "^1.0.0",
     "stats-webpack-plugin": "^0.6.1",
-    "webpack": "^3.5.5",
+    "uglifyjs-webpack-plugin": "^1.2.4",
+    "vue-loader": "^14.2.2",
+    "vue-style-loader": "^4.1.0",
+    "vue-template-compiler": "^2.5.16",
+    "webpack": "^4.5.0",
```
Comment (davidlehn) [inline package.json]: I think the webpack 4 api changed and broke some of the advanced command line flags?  Can fix that later if needed.

**Comment by dlongley** [comment]
@davidlehn,

> I think it doesn't have any overhead if you don't use .vue files?

None that I'm aware of.

File: package.json (hunk 2 above)
Comment (dlongley) [reply package.json]: Later sounds good.

File: lib/index.js (hunk 1 above)
Comment (dlongley) [reply lib/index.js]: Yes, but I don't think we're addressing it right now.

**Pull request merged**